src/logic/config/store/middleware/index.ts

```diff
@@ -1,20 +1,26 @@
 import { Action } from 'redux'
+import { _setChainId } from 'src/config'
 
 import { clearCurrentSession } from 'src/logic/currentSession/store/actions/clearCurrentSession'
 import loadCurrentSessionFromStorage from 'src/logic/currentSession/store/actions/loadCurrentSessionFromStorage'
 import { clearSafeList } from 'src/logic/safe/store/actions/clearSafeList'
 import loadSafesFromStorage from 'src/logic/safe/store/actions/loadSafesFromStorage'
 import { Dispatch } from 'src/logic/safe/store/actions/types'
 import { CONFIG_ACTIONS } from '../actions'
+import { currentChainId } from '../selectors'
+import { store as reduxStore } from 'src/store'
 
 export const configMiddleware =
-  ({ dispatch }) =>
+  ({ dispatch, getState }: typeof reduxStore) =>
   (next: Dispatch) =>
   async (action: Action<typeof CONFIG_ACTIONS.SET_CHAIN_ID>) => {
     const handledAction = next(action)
 
+    const state = getState()
     switch (action.type) {
       case CONFIG_ACTIONS.SET_CHAIN_ID: {
+        _setChainId(currentChainId(state))
```

Comment:
In which situation is this needed?

Reply:
This is our local variable that needs to be kept in line. Do you think it should be moved back into `setChainId()`?

Reply:
Oh, you moved it here. That one was nice in `setChainId` IMHO, since it's directly related to chainId. The rest of the calls are side-effect subscriptions.

Reply:
I've moved it back to `setChainId`. In hindsight, it does make more sense to remove its reliance from the store. Thanks!